src/components/AddItem/AddItem.st.css

```diff
@@ -0,0 +1,198 @@
+    DefaultTextColor: color-8;
+    DefaultTextFont: "{theme: 'Body-M', size: '14px', lineHeight: '24px'}";
+
+    DisabledBorderColor: color-3;
+    DisabledBackgroundColor: color-1;
+    DisabledTextColor: color-3;
+}
+
+/*Overrides*/
+:vars {
+    BorderColor: --overridable;
+    BackgroundColor: --overridable;
+    TextColor: --overridable;
+    TextFont: --overridable;
+}
+
+/*Fallbacks*/
+:vars {
+    CurrentBorderColor: color(fallback(value(BorderColor), value(DefaultBorderColor)));
+    CurrentBackgroundColor: color(fallback(value(BackgroundColor), value(DefaultBackgroundColor)));
+    CurrentTextColor: color(fallback(value(TextColor), value(DefaultTextColor)));
+    CurrentTextFont: fallback(font(value(TextFont)), font(value(DefaultTextFont)));
+}
+
+.root {
+    width: 100%;
+    height: 100%;
+}
+
+.wsrAddItemRoot {
+    -st-extends: AddItem;
+    -st-states: rtl, hasError, direction(enum(horizontal, vertical)), noChildren;
+}
+
+.wsrAddItemRoot:theme(dashes) {
+    border-color: value(CurrentBorderColor);
+    background-color: value(CurrentBackgroundColor);
+}
+
+.wsrAddItemRoot:theme(dashes):hover {
+    background-color: applyOpacity(value(CurrentBorderColor), 0.1);
+    border-color: applyOpacity(value(CurrentBorderColor), 0.6);
+}
+
+.wsrAddItemRoot:theme(dashes):active {
+    background-color: value(CurrentBackgroundColor);
+}
+
+.wsrAddItemRoot:hasError {
+    background-color: #FFFFFF;
+    border-color: value(errorColor);
+    opacity: 1;
+}
+
+.wsrAddItemRoot:hasError:not(:disabled):theme(dashes):hover,
+.wsrAddItemRoot:not(:disabled):focus-visible:theme(dashes):hasError {
+    background-color: applyOpacity(value(errorColor), 0.1);
+    border-color: applyOpacity(value(errorColor), 0.6);
+}
+
+.wsrAddItemRoot:theme(dashes):disabled {
+    background-color: color(value(DisabledBackgroundColor));
+    border-color: color(value(DisabledBorderColor));
+    opacity: 1;
+}
+
+/* Remove WSR focus, opacity is just for now */
+.wsrAddItemRoot:not(:disabled):focus-visible:theme(dashes) {
+    box-shadow: none;
+    border-color: value(CurrentBorderColor);
+    background-color: applyOpacity(value(CurrentBorderColor), 0.1);
+}
+
+/* direction (vertical / horizontal) */
+
+.wsrAddItemRoot:direction(vertical)::content {
+    flex-direction: column;
+}
+
+.wsrAddItemRoot:direction(horizontal)::content {
+    flex-direction: row;
+}
+
+.wsrAddItemRoot:direction(vertical)::content .icon {
+    flex-shrink: 1;
+}
+
+.wsrAddItemRoot:direction(horizontal)::content .icon {
+    flex-shrink: 0;
+}
+
+.wsrAddItemRoot:direction(vertical)::textWrapper {
+    width: 100%;
+    padding-top: 6px;
+}
+
+.wsrAddItemRoot:direction(vertical):noChildren::textWrapper {
+    padding-top: 0;
+}
+
+.wsrAddItemRoot:not(:rtl):direction(horizontal)::textWrapper {
+    max-width: 100%;
+    width: auto;
+    flex-basis: auto;
+    padding: 0 0 0 8px;
+}
+
+.wsrAddItemRoot:rtl:direction(horizontal)::textWrapper {
+    padding: 0 8px 0 0;
+}
```

Comment:
Just a general comment
Usually I prefer to avoid using the `rtl` prop in the stylesheet.

So for instance, instead of using a different padding for each direction, we usually do something like:

```css
.text::before {
  width: 8px;
  ...
}
```
and then `flex` just takes it from there.

In this specific case, it might not work, since we need to access WSR's state